controllers/controller_manager_setup.go

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	addonapiv1 "github.com/keikoproj/addon-manager/api/addon"
+	"github.com/keikoproj/addon-manager/pkg/addon"
+	"github.com/keikoproj/addon-manager/pkg/common"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func New(mgr manager.Manager) error {
+	versionCache := addon.NewAddonVersionCacheClient()
+	dynClient := dynamic.NewForConfigOrDie(mgr.GetConfig())
+	nsInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, addonapiv1.AddonResyncPeriod, addonapiv1.ManagedNameSpace, nil)
```

Comment:
Only the managed namespace ? 
The namespace of each of the addons would be watched by the controller also.

Reply:
This is only for the WorkflowInformer below, so it's restriced to addon-manager-system ns.

Reply:
I would assuming the watched namespace solution could be generic that applied to all resources that add-on manager is watching.

Reply:
I tried the manager namespace and it failed to sync caches, that can be looked into later for now we restricted this informer and also gave the controller cluster scope watch permissions just so we can unblock.